### PR TITLE
support multi-inherit

### PR DESCRIPTION
**Pull request by jycor**
syntax: https://github.com/dolthub/vitess/pull/375

### CODE REVIEW

**Review by zachmu — COMMENTED** [review]
See comments on Doltgres PR, I would just fix this to work correctly now if possible

File: sql/planbuilder/ddl.go [hunk 1]
```diff
@@ -420,13 +475,13 @@ func (b *Builder) buildCreateTableLike(inScope *scope, ct *ast.DDL) *scope {
...
-		Comment:   likeTable.Comment(),
+		Collation: coll,
+		Comment:   comment,
 	}
 
-	database := b.resolveDbForTable(ct.Table)
-
 	b.qFlags.Set(sql.QFlagSetDatabase)
+
+	outScope.setTableAlias(newTableName)
```
Comment (zachmu) [inline sql/planbuilder/ddl.go:484]: I think this is probably incorrect strictly speaking, as you'll only have the columns in the last table in the list in this scope. Not sure when it matters though. @max-hoffman ?

**Review by zachmu — APPROVED** [review]
LGTM but Max should take a look at that outscope, it's suspicious.

File: sql/planbuilder/ddl.go [hunk 2]
```diff
@@ -331,85 +331,136 @@ func assignColumnIndexesInSchema(schema sql.Schema) sql.Schema {
...
-			checkConstraint.Name = ""
-			checkDefs = append(checkDefs, checkConstraint)
+			// New check name will be assigned a name during building.
+			check.Name = ""
+		}
+		checkDefs = append(checkDefs, newCheckDefs...)
+	}
+
+	var hasSkippedCols bool
+	for _, col := range pkSch.Schema {
```
Comment (max-hoffman) [inline sql/planbuilder/ddl.go:452]: is explicit schema not incompatible with the LIKE list?

File: sql/planbuilder/ddl.go [hunk 3]
```diff
@@ -331,85 +331,136 @@ func assignColumnIndexesInSchema(schema sql.Schema) sql.Schema {
...
+			newSch = append(newSch, &newCol)
 		}
 
-		for _, check := range checks {
-			checkConstraint := b.buildCheckConstraint(outScope, &check)
-			if err != nil {
-				b.handleErr(err)
+		// if a column was skipped due to duplicates, don't copy over PK ords, idxDefs, or checkDefs
+		// since they might be incorrect
+		if hasSkippedCols {
```
Comment (max-hoffman) [inline sql/planbuilder/ddl.go:427]: are the semantics really this simple? seems like you'd need way more testing to figure out correct behavior

File: sql/planbuilder/ddl.go (hunk 2 above)
Comment (max-hoffman) [reply sql/planbuilder/ddl.go:452]: do we have tests for this?

File: sql/planbuilder/ddl.go (hunk 1 above)
Comment (max-hoffman) [reply sql/planbuilder/ddl.go:480]: it's definitely an abuse of the model...a more reasonable pass might accumulate the table cols into a top-level output scope or return an empty scope

**Review by max-hoffman — APPROVED** [review]
I still find several things suspicious and untested, it's possible they are correct for postgres and tested elsewhere. If I had to fix a bug in this code the smaller functions help, but semantically speaking it's still a little vague what should be supported

**Pull request merged**